### PR TITLE
Fix template literals with numbers

### PR DESCRIPTION
**Pull request by uurien**
### What does this PR do?

Fixes some corner cases when template literals are used with numbers

### Motivation
Fix the bugs

### Checklist

- [ ] The CHANGELOG.md has been updated
- [x] Unit tests have been updated and pass
- [ ] If known, an appropriate milestone has been selected

### CODE REVIEW

File: test/template_literal.spec.js [hunk 1]
```diff
@@ -4,36 +4,37 @@
  **/
 /* eslint-disable no-template-curly-in-string */
 /* eslint-disable no-multi-str */
-const { rewriteAndExpectNoTransformation, rewriteAndExpect } = require('./util')
+const { rewriteAst, rewriteAndExpectNoTransformation, rewriteAndExpect } = require('./util')
```
Comment (uurien) [inline test/template_literal.spec.js:7]: Most of the changes in this file are only tabs, I'll add comments in changed points

File: test/template_literal.spec.js [hunk 2]
```diff
@@ -52,170 +53,287 @@ _ddiast.plusOperator('<html lang="en">\\n    <body>\\n        <h1>XSS vulnerabil
...
-    rewriteAndExpect(js, expected)
-  })
+      rewriteAndExpect(js, expected)
+    })
 
-  it('end', () => {
-    const js = 'const result = `Hello World!${a}`;'
-    rewriteAndExpect(js, '{\nconst result = _ddiast.plusOperator("Hello World!" + a, "Hello World!", a);\n}')
-  })
+    it('Only vars', () => {
```
Comment (uurien) [inline test/template_literal.spec.js:59]: New test here

File: test/template_literal.spec.js [hunk 3]
```diff
@@ -52,170 +53,287 @@ _ddiast.plusOperator('<html lang="en">\\n    <body>\\n        <h1>XSS vulnerabil
...
+      const js = "const a = `Hello ${c} ${'how are u ' + `${'bye ' + d}`} world`;"
+      rewriteAndExpect(
+        js,
+        '{\nlet __datadog_test_0, __datadog_test_1, __datadog_test_2, __datadog_test_3, __datadog_test_4\
 , __datadog_test_5, __datadog_test_6;\n\
 const a = (__datadog_test_6 = (__datadog_test_4 = c, __datadog_test_5 = (__datadog_test_3 = (__datadog_test_2 = (\
-__datadog_test_1 = (__datadog_test_0 = _ddiast.plusOperator(\'bye \' + d, \'bye \', d), _ddiast.plusOperator("" + \
-__datadog_test_0, "", __datadog_test_0)), _ddiast.plusOperator(\'how are u \' + __datadog_test_1, \'how are u \', \
+__datadog_test_1 = (__datadog_test_0 = _ddiast.plusOperator(\'bye \' + d, \'bye \', d), _ddiast.plusOperator(\
+__datadog_test_0 + "", __datadog_test_0, "")), _ddiast.plusOperator(\'how are u \' + __datadog_test_1, \'how are u \', \
```
Comment (uurien) [inline test/template_literal.spec.js:166]: Small change in these lines, but nothing important, only the order from `"" + __datadog_test_0` to `__datadog_test_0 + ""`

File: test/template_literal.spec.js [hunk 4]
```diff
@@ -52,170 +53,287 @@ _ddiast.plusOperator('<html lang="en">\\n    <body>\\n        <h1>XSS vulnerabil
...
+        '{\nlet __datadog_test_0, __datadog_test_1, __datadog_test_2;\n\
         const a = (__datadog_test_2 = (__datadog_test_0 = a, __datadog_test_1 = new B(), _ddiast.plusOperator(\
 __datadog_test_0 + __datadog_test_1, __datadog_test_0, __datadog_test_1)), _ddiast.plusOperator("Hello " + \
 __datadog_test_2, "Hello ", __datadog_test_2));\n}'
-    )
+      )
+    })
+  })
+
+  describe('Execution tests', () => {
```
Comment (uurien) [inline test/template_literal.spec.js:234]: All the tests in this `describe` are new

File: src/transform/template_transform.rs [hunk 5]
```diff
@@ -45,16 +43,21 @@ fn get_reversed_arguments(tpl: &Tpl) -> Vec<Expr> {
...
     for quasi in &tpl.quasis {
         let value = quasi.cooked.clone();
         if value.is_none() || value.unwrap() == empty_quasi {
             if !quasi.tail {
                 let expr = &*tpl.exprs[index];
                 arguments.push(expr.clone());
                 index += 1;
+                last_skipped = true;
+            }
+            if !quasi.tail || !last_skipped {
```
Comment (simon-id) [inline src/transform/template_transform.rs:56]: so this means if either current or previous iteration `quasi.tail` is false, correct ?

**Review by simon-id — APPROVED** [review]
